crates/query-group-macro/src/queries.rs

```diff
@@ -282,11 +282,16 @@ impl ToTokens for Intern {
         let interned_pat = ty.first().expect("at least one pat; this is a bug");
         let interned_pat = &interned_pat.pat;
 
-        let wrapper_struct = self.interned_struct_path.to_token_stream();
+        let mut type_constructor = proc_macro2::TokenStream::new();
+        self.interned_struct_path.leading_colon.to_tokens(&mut type_constructor);
+        for segment in &self.interned_struct_path.segments {
+            segment.ident.to_tokens(&mut type_constructor);
+            quote! {::}.to_tokens(&mut type_constructor);
+        }
 
         let method = quote! {
             #sig {
-                #wrapper_struct::new(self, #interned_pat)
+                #type_constructor new(self, #interned_pat)
```

Comment:
Previously, when given the type `InternedOpaqueTyId<'db>` this would generate `InternedOpaqueTyId<'db>::new`, which is invalid syntax. Now it removes all the generics, leaving them to be inferred, producing just `InternedOpaqueTyId::new`. The same applies below.